src/views/blockDetails/blockBody/index.tsx

```diff
@@ -0,0 +1,236 @@
+/* eslint-disable react/jsx-no-bind */
+import { useToggleVisibility } from '@pivanov/use-toggle-visibility';
+import { formatDistanceToNowStrict } from 'date-fns';
+import { useState } from 'react';
+
+import { Icon } from '@components/icon';
+import { ModalJSONViewer } from '@components/modals/modalJSONViewer';
+import { cn } from '@utils/helpers';
+
+import type {
+  IMappedBlockBody,
+  IMappedBlockEvent,
+  IMappedBlockExtrinsic,
+} from '@custom-types/block';
+
+interface BlockBodyProps {
+  bodyData: IMappedBlockBody;
+  blockNumber: number;
+}
+
+interface ButtonProps {
+  children: React.ReactNode;
+  isActive: boolean;
+  type: string;
+  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+const Button = (props: ButtonProps) => {
+  const { children, isActive, type, onClick } = props;
+
+  return (
+    <button
+      data-filter-type={type}
+      onClick={onClick}
+      className={cn(
+        'px-2 py-2.5',
+        'font-geist font-body2-regular',
+        'duration-300 ease-in',
+        'border-b-2 border-b-transparent',
+        'hover:border-dev-pink-500 ',
+        'dark:text-dev-white-200',
+        isActive && 'border-b-dev-pink-500',
+      )}
+    >
+      {children}
+    </button>
+  );
+};
+
+export const BlockBody = (props: BlockBodyProps) => {
+  const { bodyData, blockNumber } = props;
+  const [JSONViewerModal, toggleVisibility] = useToggleVisibility(ModalJSONViewer);
+  const [modalData, setModalData] = useState<IMappedBlockEvent | IMappedBlockExtrinsic | null>(null);
+  const [filter, setFilter] = useState<'extrinsics' | 'events'>('extrinsics');
+  const [showMore, setShowMore] = useState(false);
+
+  const handleOpenModal = (data: IMappedBlockEvent | IMappedBlockExtrinsic) => {
+    setModalData(data);
+    toggleVisibility();
+  };
+
+  const handleFilter = (e: React.MouseEvent<HTMLButtonElement>) => {
+    const filterType = e.currentTarget.getAttribute('data-filter-type') as 'extrinsics' | 'events';
+    if (filterType) {
+      setFilter(filterType);
+      setShowMore(false);
+    }
+  };
+
+  const toggleShowMore = () => {
+    setShowMore(!showMore);
+  };
+
+  const visibleExtrinsics = showMore ? bodyData.extrinsics : bodyData.extrinsics.slice(0, 2);
+  const visibleEvents = showMore ? bodyData.events : bodyData.events.slice(0, 3);
+
+  const isMoreContentAvailable =
```

Comment:
we have `const [showMore, setShowMore] = useState(false);` ... we can use showMore with:
```filter === 'extrinsics' && bodyData.extrinsics.length > 2) || (filter === 'events' && bodyData.events.length > 3);`
and have one thing responsible for this job. Even more ... the only place who can eventually set `setShowMore` to true is `toggleShowMore` and the hole logic is so complicated for a simple task :)